op-service/dial/active_rollup_provider.go

```diff
@@ -105,10 +100,23 @@ func (p *ActiveL2RollupProvider) shouldCheck() bool {
 	return time.Now().After(p.activeTimeout)
 }
 
+func (p *ActiveL2RollupProvider) currentEndpoint() string {
+	if p.rollupIndex < 0 {
+		return "no endpoint set"
+	}
+	return p.rollupUrls[p.rollupIndex]
+}
+
 func (p *ActiveL2RollupProvider) findActiveEndpoints(ctx context.Context) error {
+	if p.currentRollupClient == nil {
+		err := p.dialNextSequencer(ctx)
+		if err != nil {
+			return fmt.Errorf("dialing first sequencer: %w", err)
+		}
+	}
 	for range p.rollupUrls {
 		active, err := p.checkCurrentSequencer(ctx)
-		ep := p.rollupUrls[p.rollupIndex]
+		ep := p.currentEndpoint()
 		if err != nil {
```

Comment:
I see what happens here. We only need the endpoint string in logs. What we could do instead, is have `checkCurrentSequencer()` return `false, errSeqUnset` if the current one is `nil`, where `var errSeqUnset = errors.New("sequencer unset")`, and then do here

```
if errors.Is(err, errSeqUnset) {
    log.Debug("Current sequencer unset.")
} else if ep := p.rollupUrls[p.rollupIndex]; err != nil {
// continue remaining as it is
```

This has the advantage that
- we can remove the somewhat odd `currentEndpoint` function,
- the logs are even clearer,
- we can remove the duplicate dial above the loop

Reply:
The way this is currently structured, we still need the duplicate dial, or else tests like `TestEndpointProvider_SelectSecondSequencerIfFirstInactiveAtCreation` will fail. The provider successfully dials the second sequencer, but then the loop is exhausted, so it doesn't `checkActive`. Looking into how to restructure the loop to handle that case.

Reply:
Made a readability change but the situation stands: with the new design, where `p.currentRollupClient` is initially `nil` and then initialized through a call to `RollupClient()`, we need to call `dialNextSequencer` one time more than the loop dials. But, this extra dial only needs to occur in the case where `p.currentRollupClient` is `nil`.

Added a many-sequencers, only-last-is-active pair of tests, too. They pass with the duplicate `dial`, and fail without it.

Reply:
Moved the one-time-extra-dial to the constructor, it's a bit cleaner there, I think.